README.md

```diff
@@ -324,6 +324,24 @@ Be aware that the CMake maintains a cache of build-time arguments and it will ig
 to CMake if is already in the cache.  So, between runs you need to clear the cache or even better,
 entirely remove the `build/` folder (ie. `git clean -xdf`).
 
+### Python development mode
+
+When you run `python setup.py install --user --force` it will copy python scripts into `build/Release/distr/src` and deploy as package into user site-packages (on linux in `/home/.local/`).
+To avoid deploying there use "development mode":
```

Comment:
I thought `setup.py develop` is well known, but guess we can add it to the readme..

Reply:
hmm i didn't know that, but my knowledge is limited in this.. so that is why i added. But if you think it is basics then we can skip this, to not overload README with basic stuff

Reply:
no problem, let's hear from @dkeeney

Reply:
First, keep in mind that I am not a Python programmer.  So I don't know about development mode or what it does.

I know that --user will put the package in the local site-packages so it would still load if someone is running on a shared machine and not using a virtual environment, where they may not have permissions to the global site-packages.

I 'think' that --force would cause a replace if the package is already there.

I would appreciate having someone that knows what they are talking about write what should be described in the README.  I am sure there are people that know even less about Python that I that may be trying to use our library.

Reply:
I think in some time we could split readme to subsections to avoid having one large page (partly it is i assume), in that case, more doc means better.

Reply:
> So I don't know about development mode or what it does. 

In short, "with python setup.py develop, any changes in Py code are instantly 'available' in your code when you run it.". 
about develop 
https://setuptools.readthedocs.io/en/latest/setuptools.html#develop-deploy-the-project-source-in-development-mode 
 
> some time we could split readme to subsections to avoid having one large page  

good point :+1: 
Ok, let's merge it now and I like the idea of splitting the Readme sometime into more detailed sections. (I guess "cluttering" the readme was the reason I wasn't entirely in favor of this. Since the doc isn't about anything htm specific, but about generic python usage)

Reply:
Since I was involved with changes to setup.py, our development mode might not work quite as expected. 

I do know it does have a problem in that if I change a C++ module it will not rebuild it unless I delete one of the .pyd targets. It is because there is a test to see if the targets exist and if they are all there then skip the CMake.  I keep thinking I should fix that but nobody else seems to notice. :)